microservices/playground_refresh/rds_operations.py

```
@@ -0,0 +1,175 @@
+
+    def get_latest_snapshot(self, db_instance_id: str) -> dict:
+        """Get the most recent automated snapshot for a given RDS instance."""
+        response = self.client.describe_db_snapshots(
+            DBInstanceIdentifier=db_instance_id,
+            SnapshotType="automated",
+        )
+        snapshots = response.get("DBSnapshots", [])
+        if not snapshots:
+            raise RuntimeError(
+                f"No automated snapshots found for instance {db_instance_id}"
+            )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)